server/api/controllers/praises/get-praises.js

```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  friendlyName: 'Get praises',
+
+  description: 'Get all praises',
+
+  inputs: {
+    category: {
+      description: 'category of praises to be returned',
+      type: 'string',
+      required: true,
+    },
+  },
+
+  exits: {
+    nonSuccess: {
+      description: 'Error retrieving praises.',
+    },
+    success: {
+      description: 'Successfully retrieved praises.',
+    },
+  },
+
+  fn: async function ({ category }, exits) {
+    try {
+      let data = await Praises.find();
+      if (category) {
+        sails.log.info(`Getting praise card with category: ${category}`);
+        data = data.filter((d) => d.category === category);
+      }
+
+      if (data === null) return exits.error('no data retrieved');
```

Comment:
this check should happen between line 25-26. if data === null, it would error out in line 28 before even reaching here

Reply:
willl try fix by tonight